core/src/apps/cardano/helpers/address_credential_policy.py

```diff
@@ -0,0 +1,195 @@
+
+if False:
+    from trezor.messages import CardanoAddressParametersType
+    from ..seed import Keychain
+
+
+CREDENTIAL_POLICY_SHOW = 0
+CREDENTIAL_POLICY_SHOW_KEY_PATH = 1 << 1
+CREDENTIAL_POLICY_SHOW_CHANGE = 1 << 2
+CREDENTIAL_POLICY_SHOW_OUTPUT = 1 << 3
+CREDENTIAL_POLICY_SHOW_ADDRESS = 1 << 4
+CREDENTIAL_POLICY_WARN_SCRIPT = 1 << 5
+CREDENTIAL_POLICY_WARN_POINTER = 1 << 6
+CREDENTIAL_POLICY_WARN_REWARD_ADDRESS = 1 << 7
+CREDENTIAL_POLICY_WARN_NO_STAKING = 1 << 8
+CREDENTIAL_POLICY_FAIL_OR_WARN_UNUSUAL = 1 << 9
+CREDENTIAL_POLICY_WARN_MISMATCH = 1 << 10
+
+CREDENTIAL_POLICIES_WARN = (
+    CREDENTIAL_POLICY_WARN_SCRIPT,
+    CREDENTIAL_POLICY_WARN_POINTER,
+    CREDENTIAL_POLICY_WARN_REWARD_ADDRESS,
+    CREDENTIAL_POLICY_WARN_NO_STAKING,
+    CREDENTIAL_POLICY_FAIL_OR_WARN_UNUSUAL,
+    CREDENTIAL_POLICY_WARN_MISMATCH,
+)
+
+ADDRESS_POLICY_SHOW_SPLIT = 0
+ADDRESS_POLICY_SHOW_SIMPLE = 1
+
+
+def get_address_policy(
+    keychain: Keychain, address_parameters: CardanoAddressParametersType
+) -> int:
+    if (
+        address_parameters.address_type == CardanoAddressType.BASE
+        and SCHEMA_PAYMENT.match(address_parameters.address_n)
+        and _do_base_address_credentials_match(
+            keychain,
+            address_parameters.address_n,
+            address_parameters.address_n_staking,
+            address_parameters.staking_key_hash,
+        )
+    ):
+        return ADDRESS_POLICY_SHOW_SIMPLE
+
+    return ADDRESS_POLICY_SHOW_SPLIT
+
+
+def get_change_output_payment_credential_policy(
+    address_parameters: CardanoAddressParametersType,
+    signing_mode: CardanoTxSigningMode,
+) -> int:
+    credential_policy = get_payment_credential_policy(address_parameters)
+    return _policy_to_change_output_policy(credential_policy, signing_mode)
+
+
+def get_change_output_stake_credential_policy(
+    keychain: Keychain,
+    address_parameters: CardanoAddressParametersType,
+    signing_mode: CardanoTxSigningMode,
+) -> int:
+    credential_policy = get_stake_credential_policy(keychain, address_parameters)
+    return _policy_to_change_output_policy(credential_policy, signing_mode)
+
+
+def get_payment_credential_policy(
+    address_parameters: CardanoAddressParametersType,
+) -> int:
+    if address_parameters.address_type in (
+        CardanoAddressType.BASE,
+        CardanoAddressType.BASE_KEY_SCRIPT,
+        CardanoAddressType.POINTER,
+        CardanoAddressType.ENTERPRISE,
+        CardanoAddressType.BYRON,
+    ):
+        if not SCHEMA_PAYMENT.match(address_parameters.address_n):
+            return CREDENTIAL_POLICY_FAIL_OR_WARN_UNUSUAL
+
+        return CREDENTIAL_POLICY_SHOW_KEY_PATH
+    elif address_parameters.address_type in (
+        CardanoAddressType.BASE_SCRIPT_KEY,
+        CardanoAddressType.BASE_SCRIPT_SCRIPT,
+        CardanoAddressType.POINTER_SCRIPT,
+        CardanoAddressType.ENTERPRISE_SCRIPT,
+    ):
+        return CREDENTIAL_POLICY_WARN_SCRIPT
+    elif address_parameters.address_type in (
```

Comment:
I'd add blank lines before each elif to improve readability.

Reply:
https://github.com/vacuumlabs/trezor-firmware/pull/66/commits/4d77f4e42747348ce6de09ab62a8e5ae49c07e53#diff-7774760cef1ad4636fcc2daf980fad9f3eb54dd16ef6cadeba563b245169067aR82